pkg/server/formsapi/handlers/foo/create.go

```diff
@@ -13,38 +13,65 @@ import (
 	"go.uber.org/zap"
 )
 
+func ReadFoo(req *http.Request, l *zap.Logger) (*types.FooReads, error) {
```

Comment:
If you have optional fields, you could use a pattern quite commonly used. Basically definition "option functions" for your `Foo` and passing them as a vararg parameter.

```
type FooOption func(foo *types.FooReads) *types.FooReads
var ValidFoo(isValid bool) FooOption = func(foo *types.FooReads) *types.FooReads {
  foo.valid = isValid
  return foo
}

func NewFoo(id uuid.UID, name string, options ...FooOption) *types.FooReads{
  foo := &types.FooReads{ name: name}
  for _, option := range options {
    foo = option(foo)
  }
  return foo
}

# otherpkg

var foo = NewFoo(uuid.NewV4(), "name", types.ValidFoo(false))

```

But you still have the problem that if a package function expects a `Foo`, nothing prevents the user from bypassing your constructor altogether.
```
package mypkg

function StoreFoo(foo *types.Foo) {...}

package myotherpkg

...
mypkg.StoreFoo(&types.Foo{}) // <- bypassed constructor
```

I think the only way you could really enforce this would be to use interfaces and not export the struct...

```
type Foo interface {
  GetName() string
  GetValid() bool
}

type foo struct{
 name string
 valid bool
}

func (f foo) GetName() string { return f.name}

func (f foo) GetValid() string { return f.valid}

func NewFoo(name string, valid string) Foo {
  return &foo{name: name, valid: valid}
}

```